schema/deploy/shipping/views.sql

```diff
@@ -2131,4 +2147,170 @@ grant select
+        /*
+          Keep only the most recent push. According to Lea, samples are only
+          retested if there is a failed result. A positive, negative, or
+          indeterminate result would not be retested.
+
+          https://seattle-flu-study.slack.com/archives/CV1E2BC8N/p1584570226450500?thread_ts=1584569401.449800&cid=CV1E2BC8N
+        */
+        order by
+            sample_id, presence_absence_id desc
+    ),
+
+    scan_hcov19_results as (
+      select
+          case
+              when hcov19_present is true then 'positive'
+              when hcov19_present is false then 'negative'
+              when hcov19_present is null then 'inconclusive'
+          end as hcov19_result,
+          hcov19_result_release_date,
+          case
+              when puma in ('5310901', '5310902') then 'yakima'
+              when puma in ('5311701', '5311702', '5311703',
+                            '5311704', '5311705', '5311706') then 'snohomish'
+              when puma in ('5311601', '5311602', '5311603', '5311604',
+                            '5311605', '5311606', '5311607', '5311608',
+                            '5311609', '5311610', '5311611', '5311612',
+                            '5311613', '5311614', '5311615', '5311616') then 'king'
+          end as county
+
+      from shipping.scan_encounters_v1
+      join hcov19_presence_absence using (sample)
+
+    )
+
+    select
+        hcov19_result_release_date,
+        count(*) filter (where hcov19_result in ('positive', 'inconclusive')) as total_hcov19_positives,
+        count(*) filter (where hcov19_result in ('positive', 'inconclusive') and county = 'king') as king_county_positives,
+        count(*) filter (where hcov19_result in ('positive', 'inconclusive') and county = 'snohomish') as snohomish_county_positives,
+        count(*) filter (where hcov19_result in ('positive', 'inconclusive') and county = 'yakima') as yakima_county_positives,
+        count(*) filter (where hcov19_result in ('positive', 'inconclusive') and county is null) as other_positives
+    from scan_hcov19_results
+    group by hcov19_result_release_date
+;
+
+comment on view shipping.scan_hcov19_positives_v1 is
+  'A view of counts of hcov19 positives from the SCAN project grouped by date results were released.';
+
+-- Even if it's just aggregate counts of hcov19 positives,
+-- we should probably restrict access to this view to only hcov19-visibility
+-- and power-bi-exporter.
+--  -Jover, 9 July 2020
+revoke all on shipping.scan_return_results_v1 from reporter;
+
+grant select
+    on shipping.scan_return_results_v1
+    to "hcov19-visibility";
+
+revoke all
+    on shipping.scan_hcov19_positives_v1
+  from "power-bi-exporter";
+
+grant select
+    on shipping.scan_hcov19_positives_v1
+    to "power-bi-exporter";
+
+
+create or replace view shipping.seattle_neighborhood_districts_v1 as
+
+    select
+        identifier,
+        st_assvg(point) as centroid,
+        st_assvg(polygon) as svg_path
```

Comment:
I am not 100% certain that [ST_AsSVG](http://postgis.net/docs/ST_AsSVG.html) is what we want to use here.

Reply:
Can Gernot give any input? Will he be able to say if this isn't right?

Reply:
I suspect Gernot/Formative want a static SVG file they can then dynamically style, rather than SVG paths in the data. But we can export SVG paths from Pg to create the static file.

It'd be good to use relative movements and also less point precision in the path data:

    st_assvg(polygon, rel=>1, maxdecimaldigits=>5)

This will reduce the path size by a bit and make SVG renders faster. The [fifth decimal place in lon/lat gives ~1.1m precision](https://kottke.org/14/09/the-precision-of-latlong-coordinates), which should be fine for large scale thematic mapping.